beacon-chain/core/epoch/precompute/attestation.go

```diff
@@ -13,14 +13,15 @@ import (
 	"github.com/prysmaticlabs/prysm/monitoring/tracing"
 	ethpb "github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1"
 	"github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1/attestation"
+	"github.com/prysmaticlabs/prysm/runtime/version"
 	"go.opencensus.io/trace"
 )
 
 // ProcessAttestations process the attestations in state and update individual validator's pre computes,
 // it also tracks and updates epoch attesting balances.
 func ProcessAttestations(
 	ctx context.Context,
-	state state.ReadOnlyBeaconState,
```

Comment:
Why change from ReadOnly state? Does this method now write or mutate the state?

Reply:
`state.Version()` is not defined in ReadOnly for some reason. Probably the correct thing to do is to move `Version()` to ReadOnly instead. Thanks for the comment

Reply:
Yes, I agree with that assessment. Move Version to the ReadOnly interface

Reply:
ready again @prestonvanloon